vendor/github.com/ethereum/go-ethereum/whisper/notifications/delivery.go

```diff
@@ -0,0 +1,136 @@
+package notifications
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/message"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv5"
+)
+
+// DeliveryState holds references to a message delivered through the network.
+type DeliveryState struct {
+	IsP2P bool
+	P2P   *whisper.P2PMessageState
+	RPC   *whisper.RPCMessageState
+}
+
+// DeliverySubscriber defines a function type for subscrubers.
+type DeliverySubscriber func(DeliveryState)
+
+// P2PDeliverySubscriber defines a function type for receiving p2p message status.
+type P2PDeliverySubscriber func(*whisper.P2PMessageState)
+
+// RPCDeliverySubscriber defines a function type for receiving rpc message status.
+type RPCDeliverySubscriber func(*whisper.RPCMessageState)
+
+// DeliveryService defines a notification implementation for listening to message status
+// events.
+type DeliveryService struct {
+	sml  sync.RWMutex
```

Comment:
What does `sml` stand for?

Reply:
`subscriber mutex lock`

Reply:
Would call it `subscriberMu` then, as `mu` is the typical field name for unspecialised mutex.

Reply:
+1 for `subMx` or `subMu` (I think we use more `-mx` suffixes now)